lib/builtin.pm

```diff
@@ -168,6 +168,31 @@ numerical argument.
 Returns the largest integer value less than or equal to the given numerical
 argument.
 
+=head2 indexed
+
+    @ivpairs = indexed(@items)
+
+Returns an even-sized list of number/value pairs, where each pair is formed
+of a number giving an index in the original list followed by the value at that
+position in it.  I.e. returns a list twice the size of the original, being
+equal to
+
+    (0, $item[0], 1, $item[1], 2, $item[2], ...)
+
+Note that unlike the core C<values> function, this function returns copies of
+its original arguments, not aliases to them. Any modifications of these copies
+are I<not> reflected in modifications to the original.
```

Comment:
I wonder if we need to say more here. For example, this will have lasting effects:
```
ref($_) && push(@$_, "foo") for indexed(@l);
```

Reply:
personally i wouldn't consider that a modification of the argument. if you put water in a glass have you modified the glass?

Reply:
It's "obvious" that modifying a reference will modify a reference, but at the same time "Any modifications of these copies
are I<not> reflected in modifications to the original." seems very much like "Your data will be unchanged".

People *will* be surprised.